### PR TITLE
Provide docs to describe running on CDH/HDP cluster.

### PR DESCRIPTION
**Pull request by pwendell**
This doc consolidates information relevant to CDH/HDP users in a single place.

### CODE REVIEW

File: docs/cdh-hdp.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+layout: global
+title: Running with Cloudera and HortonWorks Distributions
+---
+
+Spark is fully compatible with all versions of Cloudera's Distribution Including Hadoop (CDH) and
```
Comment (rxin) [inline docs/cdh-hdp.md]: I am not sure what "compatible" means here. Can we use a different word, e.g. Spark can run on top of ?

File: docs/cdh-hdp.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+the Hortonworks Data Platform (HDP). There are a few things to keep in mind when running against
+these distributions:
+
+# Compile-time Hadoop Version
+When compiling Spark, you'll need to
```
Comment (rxin) [inline docs/cdh-hdp.md]: Let's give an example here so it is obvious what to do.

File: docs/cdh-hdp.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+some Hadoop releases are binary compatible across client versions. This means the pre-built Spark
+distribution may "just work" without you needing to compile. That said, we recommend compiling with 
+the _exact_ Hadoop version you are running to avoid any compatibility errors.
+
+<table>
+  <tr valign="top">
+    <td>
+      <h3>CDH Releases</h3>
+      <table class="table" style="width:350px;">
+        <tr><th>Version</th><th>HADOOP_VERSION</th></tr>
```
Comment (rxin) [inline docs/cdh-hdp.md]: Nice table

File: docs/cdh-hdp.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+* Using dedicated set of Spark nodes in your cluster. These nodes should be co-located with your
+  Hadoop installation.
+* Running on the same nodes as an existing Hadoop installation, with a fixed amount memory and 
+  cores dedicated to Spark on each node.
+* Run Spark alongside Hadoop using a cluster resource manager, such as YARN or Mesos.
+
+These options are identical for those using CDH and HDP. Note that if you are running a YARN 
+cluster, you may still choose to run Spark on dedicated nodes. In this case, you should use 
+the `mr1` versions of HADOOP_HOME when compiling, not the YARN versions.
```
Comment (rxin) [inline docs/cdh-hdp.md]: This part is really confusing. Are you referring to running Spark against a separate YARN cluster?

File: docs/cdh-hdp.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+# Inheriting Cluster Configuration
+If you plan to read and write from HDFS using Spark, it is good to include copies of two relevant
```
Comment (rxin) [inline docs/cdh-hdp.md]: "it is good to copy relevant Hadoop configuration files to $SPARK_HOME/conf directory"

File: docs/cdh-hdp.md (hunk 4 above)
Comment (pwendell) [inline docs/cdh-hdp.md]: No - I mean to say that, let's say someone is running a YARN cluster for their MR jobs. They might still want to deploy Spark on dedicated nodes and not schedule Spark through YARN at all.

Comment (rxin) [inline docs/cdh-hdp.md]: Actually that's what I meant as well.

File: docs/cdh-hdp.md (hunk 1 above)
Comment (pwendell) [inline docs/cdh-hdp.md]: Hm - what if we said "Spark can read and write data from..."

File: docs/cdh-hdp.md (hunk 5 above)
Comment (mateiz) [inline docs/cdh-hdp.md]: You can also set HADOOP_CONF_DIR as far as I know.. I think that's cleaner.

**Comment by pwendell** [comment]
Hey @rxin could you take another look? For the initial paragraph, I actually do want to say something more general than just "it can read and write against HDFS". I want to say that overall, you can run side by side Hadoop and there aren't any major problems. This is just the most common I hear from ops people, they just want to know that it works - the statement is intentionally vague. For the YARN piece I tried to make it as understandable as possible. If you propose a more clear way to say this, I'm happy to change it.

File: docs/cdh-hdp.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+* Using dedicated set of Spark nodes in your cluster. These nodes should be co-located with your
+  Hadoop installation.
+* Running on the same nodes as an existing Hadoop installation, with a fixed amount memory and 
+  cores dedicated to Spark on each node.
+* Run Spark alongside Hadoop using a cluster resource manager, such as YARN or Mesos.
+
+These options are identical for those using CDH and HDP. Note that if you have a YARN cluster,
+but still prefer to run Spark on a dedicated set of nodes rather than scheduling through YARN, 
+use `mr1` versions of HADOOP_HOME when compiling.
```
Comment (mateiz) [inline docs/cdh-hdp.md]: I don't using the mr1 version matters if the cluster is running YARN -- I'd remove this

File: docs/cdh-hdp.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+a common location is inside of `/etc/hadoop/conf`. Some tools, such as Cloudera Manager, create
+configurations on-the-fly, but offer a mechanisms to download copies of them.
+
+There are a few ways to make these files visible to Spark:
+
+* You can copy these files into `$SPARK_HOME/conf` and they will be included in Spark's
+classpath automatically.
+* If you are running Spark on the same nodes as Hadoop _and_ your distribution includes both
+`hdfs-site.xml` and `core-site.xml` in the same directory, you can set `HADOOP_CONF_DIR` 
+in `$SPARK_HOME/spark-env.sh` to that directory.
```
Comment (mateiz) [inline docs/cdh-hdp.md]: Are there any distributions that put these in different directories?

File: docs/cdh-hdp.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
+---
```
Comment (mateiz) [inline docs/cdh-hdp.md]: Let's maybe name this file third-party-hadoop-distributions or something like that to be able to add more distributions later. The title and menu entry text are fine, but the filename should be something other than cdh-hdp.

File: docs/cdh-hdp.md (hunk 7 above)
Comment (pwendell) [inline docs/cdh-hdp.md]: Yes, with certain install modes CDH will put these in different directories.

File: docs/cdh-hdp.md (hunk 8 above)
Comment (pwendell) [inline docs/cdh-hdp.md]: Good idea, done.

File: docs/cdh-hdp.md (hunk 6 above)
Comment (pwendell) [inline docs/cdh-hdp.md]: Okay, removed.

**Comment by mateiz** [comment]
Looks good, thanks.

**Comment by mateiz** [comment]
Hey, I actually noticed a couple of bugs in this after merging this, like using HADOOP_VERSION instead of SPARK_HADOOP_VERSION. I've fixed them in this: https://github.com/mesos/spark/pull/905

**Pull request merged**